sdk/lib/opentelemetry/sdk/resources/resource.rb

```diff
@@ -15,19 +15,22 @@ class << self
 
           # Returns a newly created {Resource} with the specified attributes
           #
-          # @param [Hash{String => String, Numeric, Boolean} attributes Hash of key-value pairs to be used
+          # @param [Hash{String => String, Numeric, Boolean}] attributes Hash of key-value pairs to be used
           #   as attributes for this resource
+          # @param [String] Specifies the Schema URL that should be recorded in the emitted resource.
           # @raise [ArgumentError] If attribute keys and values are not strings
           # @return [Resource]
-          def create(attributes = {})
+          def create(attributes = {}, schema_url = '')
+            raise ArgumentError, 'Schema url must be a string' unless schema_url.is_a?(String)
+
             frozen_attributes = attributes.each_with_object({}) do |(k, v), memo|
               raise ArgumentError, 'attribute keys must be strings' unless k.is_a?(String)
               raise ArgumentError, 'attribute values must be (array of) strings, integers, floats, or booleans' unless Internal.valid_value?(v)
 
               memo[-k] = v.freeze
             end.freeze
 
-            new(frozen_attributes)
+            new(frozen_attributes, schema_url)
```

Comment:
We should `freeze` the `schema_url` as well.
```suggestion
            new(frozen_attributes, schema_url.freeze)
```